File: mocks/dokumentproduksjon-mock/src/main/java/no/nav/pensjon/vtp/mocks/fpformidling/FpFormidlingMock.java
package no.nav.pensjon.vtp.mocks.fpformidling;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import no.nav.pensjon.vtp.mocks.fpformidling.dto.*;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.*;

@RestController
@Api("/fpformidling")
@RequestMapping("/fpformidling")
public class FpFormidlingMock {
    private final Map<UUID, List<String>> dokumentProduksjon = new HashMap<>();
    private final Map<UUID, TekstFraSaksbehandlerDto> saksbehandlerTekst = new HashMap<>();

    @PostMapping(value = "/hent-dokumentmaler", produces = MediaType.APPLICATION_JSON_VALUE)
    @ApiOperation(value = "HentDokumentmalListe", notes = ("Returnerer tilgjengelige dokumentmaler"))
    public ResponseEntity hentDokumentmalListe(BehandlingUuidDto request) {
        return ResponseEntity.ok(new HentBrevmalerDto(Collections.emptyList()));
    }

    @PostMapping(value = "brev/maler", produces = MediaType.APPLICATION_JSON_VALUE)
    public HentBrevmalerDto hentBrevmaler(BehandlingUuidDto uuidDto) {
        return new HentBrevmalerDto(Collections.emptyList());
    }

    @PostMapping(value = "brev/dokument-sendt", produces = MediaType.APPLICATION_JSON_VALUE)
    public Boolean erDokumentSendt(DokumentProdusertDto request) {
        return dokumentProduksjon.getOrDefault(request.getBehandlingUuid(), List.of()).contains(request.getDokumentMal());
    }

    @PostMapping(value = "brev/bestill", produces = MediaType.APPLICATION_JSON_VALUE)
    public void bestillDokument(DokumentbestillingDto request) {
        dokumentProduksjon.putIfAbsent(request.getBehandlingUuid(), new ArrayList<>());
        dokumentProduksjon.get(request.getBehandlingUuid()).add(request.getDokumentMal());
    }

    @PostMapping(value = "saksbehandlertekst/hent", produces = MediaType.APPLICATION_JSON_VALUE)
    public TekstFraSaksbehandlerDto hentSaksbehandlersTekst(BehandlingUuidDto uuidDto) {
        return saksbehandlerTekst.getOrDefault(uuidDto.getBehandlingUuid(), null);
    }

    @PostMapping(value = "saksbehandlertekst/lagre", produces = MediaType.APPLICATION_JSON_VALUE)
    public void lagreSaksbehandlersTekst(TekstFraSaksbehandlerDto request) {
        saksbehandlerTekst.put(request.getBehandlingUuid(), request);
    }

}

File: vtp-pensjon-application/src/main/kotlin/no/nav/pensjon/vtp/mocks/fpformidling/FpFormidlingMock.kt
package no.nav.pensjon.vtp.mocks.fpformidling

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import no.nav.pensjon.vtp.mocks.fpformidling.dto.BehandlingUuidDto
import no.nav.pensjon.vtp.mocks.fpformidling.dto.DokumentProdusertDto
import no.nav.pensjon.vtp.mocks.fpformidling.dto.DokumentbestillingDto
import no.nav.pensjon.vtp.mocks.fpformidling.dto.HentBrevmalerDto
import no.nav.pensjon.vtp.mocks.fpformidling.dto.TekstFraSaksbehandlerDto
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.*
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.CopyOnWriteArrayList

@RestController
@Tag(name = "/fpformidling")
@RequestMapping("/rest/fpformidling")
class FpFormidlingMock {
    private val dokumentProduksjon: MutableMap<UUID, MutableList<String>> = ConcurrentHashMap()
    private val saksbehandlerTekst: MutableMap<UUID, TekstFraSaksbehandlerDto?> = ConcurrentHashMap()

    @PostMapping(value = ["/hent-dokumentmaler"])
    @Operation(summary = "HentDokumentmalListe", description = "Returnerer tilgjengelige dokumentmaler")
    fun hentDokumentmalListe(request: BehandlingUuidDto?) = HentBrevmalerDto(emptyList())

    @PostMapping(value = ["brev/maler"])
    fun hentBrevmaler(uuidDto: BehandlingUuidDto?): HentBrevmalerDto = HentBrevmalerDto(emptyList())

    @PostMapping(value = ["brev/dokument-sendt"])
    fun erDokumentSendt(request: DokumentProdusertDto): Boolean {
        return dokumentProduksjon.getOrDefault(request.behandlingUuid, emptyList()).contains(request.dokumentMal)
    }

    @PostMapping(value = ["brev/bestill"])
    fun bestillDokument(request: DokumentbestillingDto) {
        dokumentProduksjon.computeIfAbsent(request.behandlingUuid) { CopyOnWriteArrayList() }
            .add(request.dokumentMal)
    }

    @PostMapping(value = ["saksbehandlertekst/hent"])
    fun hentSaksbehandlersTekst(uuidDto: BehandlingUuidDto): TekstFraSaksbehandlerDto? {
        return saksbehandlerTekst.getOrDefault(uuidDto.behandlingUuid, null)
    }

    @PostMapping(value = ["saksbehandlertekst/lagre"])
    fun lagreSaksbehandlersTekst(request: TekstFraSaksbehandlerDto) {
        saksbehandlerTekst[request.behandlingUuid] = request
    }
}
